src/csv_module.f90

```diff
@@ -1217,7 +1217,7 @@ pure subroutine split(str,token,chunk_size,vals)
         call expand_vector(itokens,n_tokens,chunk_size,i+j)  ! save the token location
         if (i+len_token>len_str) exit  ! if the last bit of the string is a token
         j = j + i
-        temp = temp(i+len_token:len_str)  !remove previously scanned part of string
+        temp = trim(temp(i+len_token:len_str))  !remove previously scanned part of string
```

Comment:
What problem does this solve? What if the token is a space? I think in that case, you wouldn't want to trim the string.